cellprofiler/modules/filterobjects.py

```diff
@@ -274,6 +280,13 @@ def get_rules_class_choices(self, pipeline):
         except:
             return [str(i) for i in range(1, 3)]
         
+    def get_classifier_class_choices(self, pipeline):
```

Comment:
I think your best bet is to move the logic in `get_classifier_class_choices` into `get_rules_class_choices` (and maybe rename `get_rules_class_choices` to `get_class_choices`. You can put an if statement to test whether the mode is `MODE_RULES` or `MODE_CLASSIFIERS`.